### PR TITLE
Rhmap 17815 add component update stage fh v3.19

### PR DESCRIPTION
**Pull request by mikenairn**
Jira:

https://issues.jboss.org/browse/RHMAP-17815

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
 // https://github.com/feedhenry/fh-pipeline-library
-@Library('fh-pipeline-library') _
-
-final String COMPONENT = 'memcached'
-final String VERSION = '1.4.15'
-final String DOCKER_HUB_ORG = "rhmap"
-final String DOCKER_HUB_REPO = COMPONENT
+@Library('fh-pipeline-library@origin/pr/71') _
 
 fhBuildNode(['label': 'openshift']) {
```
Comment (grdryn) [inline Jenkinsfile]: For node module projects, we've stopped using `fhBuildNode`. Should we consider moving away from it for repos like this also?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,17 +1,35 @@
 #!groovy
 
 // https://github.com/feedhenry/fh-pipeline-library
-@Library('fh-pipeline-library') _
-
-final String COMPONENT = 'memcached'
-final String VERSION = '1.4.15'
-final String DOCKER_HUB_ORG = "rhmap"
-final String DOCKER_HUB_REPO = COMPONENT
+@Library('fh-pipeline-library@origin/pr/71') _
```
Comment (grdryn) [inline Jenkinsfile]: TODO: change this back before merging

Comment (mikenairn) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 1 above)
Comment (mikenairn) [reply Jenkinsfile]: We can, but really if we do it here, we would have to do it everywhere. Maybe we should, i'm not sure. fhBuildNode is doing the same for this as it is for any other rhmap component.

Comment (mikenairn) [reply Jenkinsfile]: Discussed offline and decided we would look at doing this later in a separate piece of work.

File: Jenkinsfile [hunk 3]
```diff
@@ -3,15 +3,34 @@
...
     stage('Build Image') {
-        dockerBinaryBuild(COMPONENT, VERSION, DOCKER_HUB_ORG, DOCKER_HUB_REPO, 'dockerhubjenkins', '.')
+        final Map params = [
+                fromDir: '.',
+                buildConfigName: COMPONENT,
+                imageRepoSecret: "dockerhub",
+                outputImage: "docker.io/${DOCKER_HUB_ORG}/${DOCKER_HUB_REPO}:${VERSION}-${BUILD}"
+        ]
+        buildWithDockerStrategy params
+        archiveArtifacts writeBuildInfo('memcached-container', "${VERSION}-${BUILD}")
```
Comment (mikenairn) [inline Jenkinsfile]: @grdryn Added this since you last reviewed, required for https://issues.jboss.org/browse/RHMAP-17650

Comment (grdryn) [reply Jenkinsfile]: 👍

**Pull request merged**